reconcile/sendgrid_teammates.py

```diff
@@ -47,28 +55,52 @@ def fetch_current_state(sg_client):
             # we want to ignore the root account (owner account)
             continue
 
-        t = Teammate(teammate['email'], pending=False)
+        t = Teammate(teammate['email'])
         state.append(t)
 
     return state
 
 
+def raise_if_error(response):
+    """
+    Raises an SendGridAPIError if the request has returned an error
+    """
+    if int(response.status_code / 100) != 2:
+        raise SendGridAPIError(response.body.decode('utf-8'))
+
+
 def act(dry_run, sg_client, desired_state, current_state):
     """
     Reconciles current state with desired state.
 
-    :return: whether there has been an error
+    :return: true if there has been an error
     :rtype: bool
     """
 
     desired_emails = [e.email for e in desired_state]
     current_emails = [e.email for e in current_state]
 
+    error = False
+
     for user in current_state:
         if user.email not in desired_emails:
             LOG.info(['delete', user.email])
+            if not dry_run:
+                if user.pending:
+                    delete_method = sg_client.teammates.pending
+                    identifier = user.pending_token
+                else:
+                    delete_method = sg_client.teammates
+                    identifier = user.email
+
+                response = delete_method._(identifier).delete()
```

Comment:
This underscore thing is tricky. I kind of inferred it from here: https://github.com/sendgrid/sendgrid-python#general-v3-web-api-usage-without-fluent-interface